### ISSUE TITLE
How to test WordPress Flexible env. locally

### ISSUE THREAD

**Issue by daron4ever** [GoogleCloudPlatform/php-docs-samples#298]
hello everyone!

I would like to test a WordPress application (Flexible Environment) locally with local database, not Cloud SQL.

I have scripted
`$ php -S localhost:8080 wordpress`
It didn't work and got the following errors:
`Warning: Unknown: failed to open stream: No such file or directory in Unknown on line 0

Fatal error: Unknown: Failed opening required 'wordpress' (include_path='.:') in Unknown on line 0`

I think the problem is in my app.yaml.
```
# application: xxxx
# version: 2017-01-10-test
service: default
runtime: php
env: flex

beta_settings:
  cloud_sql_instances: "xxxx:asia-northeast1:xxxxx"

runtime_config:
  document_root: wordpress

env_variables:
  WHITELIST_FUNCTIONS: escapeshellarg,escapeshellcmd,exec,pclose,popen,shell_exec,phpversion,php_uname
```

Please help!

**Comment by tmatsuo** [comment]
You can use the `wp` tool in the vendor directory.

$ vendor/bin/wp --path=wordpress server

I think you need to install `memcache` extension. Also run memcached server for the cache plugins to work.

If you don't need the memcache based cache, you can just remove `wordpress/wp-content/advanced-cache.php` and `wordpress/wp-content/object-cache.php`.

Also, probably you need to modify nginx-app.conf when you deploy the wordpress. Sorry about that. Please modify it to the following:

```
location / {
    try_files $uri /index.php?q=$uri&$args;
}

location ~ ^/wp-admin {
    try_files $uri $uri/index.php?$args;
}
```

I will fix the helper script as soon as possible too.

**Comment by daron4ever** [comment]
@tmatsuo ,
Thanks. It works locally well without the memcache extension.
But, I can not connect my app to Cloud SQL at live environment.

I have the DB connection error.
```
Error establishing a database connection
This either means that the username and password information in your wp-config.php file is incorrect or we can’t contact the database server at 127.0.0.1. This could mean your host’s database server is down.
```
I have checked my database information in wp-confg.php many times and it is right.

If I check my logo in App Engine:

```
{
 textPayload: "2017/01/13 13:38:40 Open socket for "xxx-xx0812:asia-northeast1:xxx-wordpress" at "/cloudsql/xxx-xx0812:asia-northeast1:xxx-wordpress"
"   
 insertId: "yyvboffajf5xx"   
 resource: {…}   
 timestamp: "2017-01-13T13:38:40Z"   
 labels: {…}   
 logName: "projects/ejusu-150812/logs/appengine.googleapis.com%2Fstderr"   
}
```

**Comment by tmatsuo** [comment]
Oh that's another issue with `env: flex`.
Open the `wordpress/wp-config.php` and look for a line defining `ON_GAE` constant, then change it to:
```
define('ON_GAE', getenv('GAE_VERSION') !== false || getenv('GAE_VM') !== false);
```

**Comment by daron4ever** [comment]
@tmatsuo ,
I changed it, but still doesn't work on the live server.

**Comment by tmatsuo** [comment]
@daron4ever What happens? Any logs? Can you invite me as an viewer of that project?

**Comment by tmatsuo** [comment]
Ah, I realized that I pasted a wrong snippet in https://github.com/GoogleCloudPlatform/php-docs-samples/issues/298#issuecomment-272481442

Maybe you can try this instead:
```
$onGae = getenv('GAE_VERSION') !== false || getenv('GAE_VM') !== false;
```

**Comment by tmatsuo** [comment]
@daron4ever 

JFYI, I have just updated the script to support `env: flex`. Can you try again and report back?

**Comment by daron4ever** [comment]
@tmatsuo I don't use this flexible env anymore. sorry!

**Comment by bld2104** [comment]
@tmatsuo I am having the same problem, and I tried updating the onGae variable like you said and it's still not working. Here's a screenshot of the error. 

<img width="755" alt="screen shot 2018-01-10 at 3 26 30 pm" src="https://user-images.githubusercontent.com/7794512/34793874-ce655e14-f61a-11e7-860e-0432be6aa4cb.png">

I'm actually also having a (possibly) similar problem with MAMP - MAMP seems to be ignoring the if(onGae) part of my wp-config and trying the production credentials. Here's my Stack Overflow question about it with more details: https://stackoverflow.com/questions/48194954/mamp-attempting-to-connect-to-production-sql-database-localhost-php-google-app

**Comment by tmatsuo** [comment]
@bld2104 What produced the error? Did you try to run the WordPress with the PHP built-in server?

**Comment by bld2104** [comment]
@tmatsuo yes as described in [my post to Stack Overflow](https://stackoverflow.com/questions/48194954/mamp-attempting-to-connect-to-production-sql-database-localhost-php-google-app), first I tried with MAMP and when that didn't work I tried with the PHP built-in server. The error displayed in the image above is with the PHP built-in server.

**Comment by bld2104** [comment]
@tmatsuo I just realized something - I set up a local database for MAMP (through MAMP's point-and-click interface) but I guess I never set one up for the built-in Google local server. In trying to google how to do this I found [an article you wrote](https://wp.gaeflex.ninja/2016/03/14/running-wordpress-on-flex/)! :) It doesn't mention setting up a local database, so should I be commenting that part of my wp-config that talks about the local database I set up using MAMP? I am absolutely fine using the Google built-in server btw, I don't need to use MAMP - I'm just trying to get some local server working consistently.

Also I know this is a different topic, but I don't have that cache thing you mention at the end of the article, should I have those plugins installed? Is that caching my site to improve its loading time?
